cert/aziot-certd-config/src/lib.rs

```diff
@@ -45,130 +53,110 @@ pub struct Config {
 }
 
 /// Configuration of how new certificates should be issued.
-#[derive(Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
 pub struct CertIssuance {
     /// Configuration of parameters for issuing certs via EST.
     pub est: Option<Est>,
 
     /// Configuration of parameters for issuing certs via a local CA cert.
-    pub local_ca: Option<LocalCa>,
+    pub local_ca: Option<CertAuthority>,
 
     /// Map of certificate IDs to the details used to issue them.
     #[serde(flatten)]
-    pub certs: std::collections::BTreeMap<String, CertIssuanceOptions>,
+    pub certs: BTreeMap<String, CertIssuanceOptions>,
 }
 
 /// Configuration of parameters for issuing certs via EST.
-#[derive(Debug, PartialEq)]
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
 pub struct Est {
-    /// Authentication parameters for the EST server.
-    pub auth: EstAuth,
-
     /// List of certs that should be treated as trusted roots for validating the EST server's TLS certificate.
+    #[serde(default)]
     pub trusted_certs: Vec<String>,
 
+    /// Authentication parameters for the EST server.
+    // NOTE: DO NOT MOVE. Tables must be after values!
+    #[serde(flatten)]
+    pub auth: EstAuth,
+
     /// Map of certificate IDs to EST endpoint URLs.
     ///
     /// The special key "default" is used as a fallback for certs whose ID is not explicitly listed in this map.
-    pub urls: std::collections::BTreeMap<String, url::Url>,
+    pub urls: BTreeMap<String, Url>,
 }
 
 /// Authentication parameters for the EST server.
 ///
 /// Note that EST servers may be configured to have only basic auth, only TLS client cert auth, or both.
-#[derive(Clone, Debug, PartialEq)]
+#[skip_serializing_none]
+#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize)]
+#[serde(try_from = "_EstAuth")]
 pub struct EstAuth {
     /// Authentication parameters when using basic HTTP authentication.
+    #[serde(flatten)]
     pub basic: Option<EstAuthBasic>,
 
     /// Authentication parameters when using TLS client cert authentication.
+    #[serde(flatten)]
+    pub x509: Option<EstAuthX509>,
+
+    /// Headers to inject into authentication request.
+    pub headers: Option<BTreeMap<String, String>>
+}
+
+#[derive(Deserialize)]
+struct _EstAuth {
```

Comment:
Leading `_` in Rust means "unused". Something like the original `EstAuthInner` or `EstAuth_` would be better.